x/il/keeper/abci.go

```diff
@@ -156,11 +159,49 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 		// for the cases when the outgoing txs to ethereum fail for some reason.
 		k.DeleteStoplossPosition(ctx, address, stoploss.UniswapPairID)
 
+		// log and emit metrics
+		k.Logger(ctx).Debug("stoploss batched", "pair", stoploss.UniswapPairID, "receiver-address", stoploss.ReceiverAddress)
+
 		return false
 	})
 
+	// return if there is no payload
+	if len(batchedPositions.Payloads) == 0 {
+		k.Logger(ctx).Debug("no payload for batched execution")
+		return
+	}
+
+	// increment the invalidation ID counter
+	invalidationID++
+
+	// encode the simple logic batch ABI
+	payload, err := batchedPositions.GetEncodedCall()
+	if err != nil {
+		panic(fmt.Errorf("sommelier contract ABI payload pack failed: %w", err))
+	}
+
+	// NOTE: by setting the invalidation nonce always to 0 and the invalidation ID to an increasing
+	// counter, will prevent a logic call to get invalidated unless the outgoing Ethereum transaction
+	// times out
+
+	call := &bridgetypes.OutgoingLogicCall{
+		Transfers:            transfers,
+		Fees:                 fees,
+		LogicContractAddress: params.BatchContractAddress, // TODO: double check
+		Payload:              payload,
+		Timeout:              ethHeight.EthereumBlockHeight + params.EthTimeoutBlocks,
+		InvalidationId:       sdk.Uint64ToBigEndian(invalidationID), // TODO: should this be hex?
+		InvalidationNonce:    0,
+	}
+
+	// send eth transaction to withdraw lp_shares liquidity for pair_id
+	k.ethBridgeKeeper.SetOutgoingLogicCall(ctx, call)
+
 	// Set the new invalidation
 	if originalInvalidationID != invalidationID {
```

Comment:
I don't exactly follow what is going on here with the invalidation ID, but it probably should not be happening. 

To keep things simple, just use a new invalidation ID every single time you call `SetOutgoingLogicCall`.

There could be some optimizations in the future where we reuse invalidation IDs to save on Eth gas, but they are not worth doing right now.